app.js

```diff
@@ -0,0 +1,60 @@
+﻿'use strict';
+const debug = require('debug');
+const express = require('express');
+const logger = require('morgan');
+const bodyParser = require('body-parser');
+const cors = require('cors');
+
+const secretStorage = require('./secret-storage.js');
+const NoSuchEntity = secretStorage.NoSuchEntity;
+const DecryptionError = secretStorage.DecryptionError;
+
+const app = express();
+
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(cors());
+app.set('port', process.env.PORT || 1234);
+
+
+const server = app.listen(app.get('port'), () => {
+    debug('Express server listening on port ' + server.address().port);
+});
+
+app.use((err, req, res, next) => {
+    console.error(err.stack);
+    res.status(500).end();
+    next();
+});
+
+app.put('/secret', (req, res) => {
+    secretStorage.put(req.body.name, req.body.secret, req.body.encryptionKey).then(res.status(201).end());
```

Comment:
Basically, the password will serve a second function to authenticate a user, and not just to authorize access to passwords